articles/tips-and-tricks-for-deep-learning/index.md

```diff
@@ -0,0 +1,252 @@
+
+##### Color shifted Image
+
+![Color shifted Image](/engineering-education/tips-and-tricks-for-deep-learning/color-shifted-image.jpg)<br>
+
+##### Contrast changed Image
+
+![Contrast changed Image](/engineering-education/tips-and-tricks-for-deep-learning/contrast-changed-image.jpg)<br>
+
+If done correctly, data augmentation can be a very powerful tool. It forces the neural network to focus more on an image's attributes instead of the image themselves. Besides, it also increases the data needed for training.
+
+#### Batch Normalization (Batch Norm)
+
+Batch normalization is also known as Batch Norm. It is a technique used in deep learning to stabilize models and speed up learning.  
+
+Before we dive into how batch normalization works, let's first briefly discuss the working of a typical normalization technique. This discussion will make you better understand why we use batch normalization.
+
+Normalization is a pre-processing step on input data that helps us put data on a standard scale. This process is an important step as there might be variations in input data. Some input data might be having higher values while some, very low values. Let's consider an example of car mileages over a ten-year period to demonstrate non-normalized data. 
+
+One of the drivers, aged 50, has covered a mileage of 100,000 while another driver, aged 25, has only covered 1,000 miles on their cars. The mileage and age data from these two drivers vary widely thus might not fit on the same scale. 
+
+Using such imbalanced data as inputs to our neural network can cause four major issues:
+1. An unstable neural network  
+2. An imbalanced gradient that leads to the exploding gradient problem. Read more about it [here](https://deepai.org/machine-learning-glossary-and-terms/exploding-gradient-problem/).
+3. It makes the network very difficult to train. 
+4. It decreases the training speed.      
+
+To avoid the problems mentioned above, we normalize their ages and mileage inputs to fit on the same scale. From our example, we can scale down our mileage and age data to a scale of between 0 to 1. Normalization reduces the wide ranges between the data points. However, normalization only solves part of the problem.
+
+From my previous [article](https://www.section.io/engineering-education/basics-of-convolution-neural-networks/), I discussed how training is achieved by updating the weights of the neural network iteratively through a process known as backpropagation. But within each iteration, these weights are updated simultaneously. This is a problem as it means that as the weights are updated, the weights of the previous layers are also updating simultaneously. This phenomenon is known as the internal covariate shift. It ends up slowing down the training as it requires the use of lower learning rates. Also, it makes it harder to train models as careful initialization of the parameters is needed. The internal covariate shift is the problem batch normalization aims to solve.
```

Comment:
The process of updation of weights is known as internal covariate shift? It's not clear what internal covariate shift is.